### DISCUSSION TITLE
Pico 2 ✨

### DISCUSSION THREAD

**Discussion by lucaslarroche** [picocss/pico#355] [General]
**The first alpha version for Pico 2 is here: `v2.0.0-alpha1`!** 

 This version significantly improves the look and feel, accessibility, and customization options. Pico v2 features a new color palette with 380 manually crafted colors, new breakpoints, and a new group component for stacking form elements and buttons horizontally.

Here are all the links to discover Pico 2 `alpha1`:

- [Online documentation](https://v2.picocss.com/docs)
- [NPM pre-release](https://www.npmjs.com/package/@picocss/pico/v/2.0.0-alpha1)
- [GitHub pre-release](https://github.com/picocss/pico/releases/tag/v2.0.0-alpha1)
- [Website code](https://github.com/picocss/picocss.com)
- [v2 branch](https://github.com/picocss/pico/tree/v2)

Feel free to comment here, or to open PRs again [v2](https://github.com/picocss/pico/tree/v2) or the [documentation](ttps://github.com/picocss/picocss.com).

# 😎 New look and feel

All colors have been carefully and manually redefined for a more contrasted, neutral (less bluish) look and feel.

The default color theme is much more accessible, with most colors now following the WCAG 2.1 AAA standard. Some secondary muted colors follow the WCAG 2.1 AA standard.

Focus states have been improved for more consistency and contrast.

# 🛠️ Easier customization

We have refactored all `.scss` files to make it easier for you to compile your own version of Pico with [SASS](https://v2.picocss.com/docs/sass). All modules can now be enabled or disabled using [@use](https://sass-lang.com/documentation/at-rules/use) and vars.

# 🎨 Color palette

Pico v2 comes with [380 manually crafted colors](https://v2.picocss.com/docs/colors) to help you personalize your brand design system. The colors can be imported into any `.scss`file, and a new stylesheet with all the color utilities is provided.

# 📐 Better breakpoints

We have updated the [breakpoints](https://v2.picocss.com/docs/container) to follow the width of standard devices and added a new breakpoint for larger screens.

# 🧩 New group component

With [group](https://v2.picocss.com/docs/group) (`role="group"`), you can now stack forms elements and buttons horizontally.

# 💥 Breaking changes

## CSS Vars

All [CSS custom properties](https://v2.picocss.com/docs/css-variables) are prefixed with `pico-`to avoid collisions with other CSS frameworks or your own vars. We also added new CSS variables and renamed some to follow a consistent pattern`{component}-{state}-{property}`.

## SCSS files

We renamed and moved many `.scss` files. If you import Pico modules, you need to update your paths.

## Buttons

[Buttons](https://v2.picocss.com/docs/button) are not `width: 100%;` by default anymore. Only form buttons are full-width to be consistent with other form elements.

## Table

The `.striped` class is now used for the striped [table](https://v2.picocss.com/docs/table)'s style.

## Dropdowns

While accordions are still classless, [dropdowns](https://v2.picocss.com/docs/dropdown) now use`.dropdown`. We also removed the experimental syntax that allowed dropdowns in the nav using nested lists.

## Grid

[Grid](https://v2.picocss.com/docs/grid) columns now collapse on small devices (`<768px`).

**Comment by lucaslarroche** [comment]
In the following weeks, I will work on the examples:
- HTML Preview
- Classless HTML Preview
- React
- Svelte
- Gatsby
- Next

**Reply by micene09** [reply]
what about Vue/Nuxt?

**Reply by bozzhik** [reply]
what about Astro?
https://astro.build/

**Comment by peteringram0** [comment]
@lucaslarroche Is V2 still on the way at all?

**Reply by lucaslarroche** [reply]
Yes. I find it difficult to find time for my work.
The library shouldn't change too much.
I still have to build the examples and the homepage.

**Comment by ralacerda** [comment]
Oh, that is a really nice release.
I was wondering if you are planning on using `:focus-visible` instead of `:focus` in clickable elements. It prevents buttons and anchors tags to keep the focus state after being clicked on, but It's not supported on IE. 

I'll probably keep a fork with that change anyway since I don't plan on supporting IE.

**Comment by jeffschwartz** [comment]
This release looks to be a real step up and thank you for this amazing library. Can you provide a timeline for its release?

**Comment by jeffschwartz** [comment]
BTW have you considered adding tabs as a component to v2? An example can be found here https://erikpoehler.com/2022/12/15/picocss-content-tabs/. It would make a great addition to Pico's components.

**Reply by asbjornu** [reply]
The project seems to accept pull requests, so I suppose it's just to submit one?

**Comment by brentarias** [comment]
I would love to see a responsive hamburger menu.  Indeed I made a [hamburger menu for PicoCSS](https://github.com/picocss/pico/issues/62#issuecomment-1666978043) v1 you could reference.

**Comment by brentarias** [comment]
Also, this might be too "radical" for inclusion in PicoCSS...though have you seen this cool ["re-working" of the idea of "containers"](https://www.youtube.com/watch?v=c13gpBrnGEw)?  I was rather impressed.

**Comment by micene09** [comment]
Hey @lucaslarroche, what a great job you did with this v2 release...I love it, simply amazing!
It's been a while since the last commit and I hope you are fine.
In the meanwhile, looking at the v2 branch, the issues (not so many and not all real issues) and the Projects board, it's not clear to me what do you need to stabilize the v2, I would like to help you somehow.

**Comment by Shreyans13** [comment]
Hey @lucaslarroche 
This release looks really good.....liked a lot of features was planning to use in my project ......when can we see a released version for this ?

**Comment by xpiu** [comment]
I'm enjoying Pico v1 classless <3 
Minor comment for v2: I would like a little extra info on the differences between _pico.classless.scss_ and _pico.fluid.classless.scss_ Perhaps there's some place for it in the Readme file.
Great work guys.

**Comment by peterbe** [comment]
Gongrats on the launch! It looks great. I'm putting 2.0 into https://www.peterbe.com as we speak.
